### PR TITLE
2624 fix enum quotes for strings

### LINKED ISSUE #2624: enum.prepare by default uždedamos kabutės jeigu type yra string
**Specifikacijos punktas:**
- https://github.com/atviriduomenys/spinta/issues/1094
#

Šiuo metu katalogas pagal nutylėjima uždeda kabutes ant enum.prepare jeigu jis yra string tipo. Reikia, kad automatiškai to nebūtų daroma, nes neapibrėžtas veikimas, kai yra tuščios kabutės arba kai pateikta funkcija.

Taip pat, galimai dėl šio bug'o, yra galima importuoti DSA, kur išvis nenurodytos enum.prepare ir enum.source reikšmės, nes jom automatiškai uždedamos kabutės.

Susije:
- https://github.com/ivpk/dsa/issues/46

### PR DESCRIPTION
**Pull request by gts446**
## Remove automatic enum quote handling in forms and fix empty-source edge case

Previously, the enum item form silently managed quotation marks on behalf
of the user: it stripped quotes before displaying a string enum value and
re-added them before saving. It also ran type validation
(string/integer/boolean) synchronously on form submit.

This PR removes that automatic behaviour so values are stored and displayed
exactly as entered. Type validation together with the rest of structure will be handled separately https://github.com/atviriduomenys/katalogas/issues/2370

## Changes

- Remove auto-stripping of quotes from string enum values in the form's
  initial data
- Remove auto-wrapping of unquoted string values in `clean_value()`
- Remove synchronous type checking (`TypeCheckerError`) from `clean_value()`
- Fix edge case in CSV import: when `prepare` is empty and `source` is also
  empty, no longer produces `""` as a valid quoted value
- Update tests accordingly

### CODE REVIEW

File: vitrina/structure/forms.py [hunk 1]
```diff
@@ -242,16 +237,6 @@ def clean_value(self):
         if not (value := self.cleaned_data.get("value")):
             return value
 
-        if metadata := self.prop.metadata.first():
-            if metadata.type == "string" and not is_quoted(value):
-                value = f'"{value}"'
-
-            checker = metadata.get_type_checker()
```
Comment (martynasracct) [inline vitrina/structure/forms.py:249]: Are these type-checkers used anywhere else? If not, we should also delete them

**Review by martynasracct — APPROVED** [review]

**Review by Struchid — APPROVED** [review]

**Review by Gustascct — APPROVED** [review]

File: vitrina/structure/forms.py (hunk 1 above)
Comment (Struchid) [reply vitrina/structure/forms.py:249]: Yup, not used anywhere else, getting rid of it:
- https://github.com/atviriduomenys/katalogas/pull/2637/changes/62feab51e540357d1301da5a7e5f4182edb2f405

**Review by MindaugasStanionis — APPROVED** [review]
